lib/core/client.js

```diff
@@ -700,10 +702,31 @@ function connect (client) {
   if (protocol === 'https:') {
     servername = client[kServerName] || (client[kTLSOpts] && client[kTLSOpts].servername)
     const tlsOpts = { ...client[kTLSOpts], servername }
+    const sessionName = createName(client[kUrl], tlsOpts)
```

Comment:
I think sessionname can be mostly precalculated in the constructor? except for `client[kServerName]` which we could simple append.

Reply:
Yeah, we could cache it but let's keep it as is for now, as I don't think it would save that much 🤔